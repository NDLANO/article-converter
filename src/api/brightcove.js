/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import fetch from 'isomorphic-fetch';
import { resolveJsonOrRejectWithError } from '../utils/apiHelpers';
import { brightcoveClientId, brightcoveClientSecret } from '../config';

function fetchVideo(videoId, accountId, accessToken) {
  const url = `https://cms.api.brightcove.com/v1/accounts/${accountId}/videos/${videoId}`;

  return fetch(url, {
    method: 'GET',
    headers: {
      'content-type': 'Content-Type: application/json',
      Authorization: `Bearer ${accessToken.access_token}`,
    },
  });
}

function getAccessToken() {
  const base64Encode = str => new Buffer(str).toString('base64');
  const url =
    'https://oauth.brightcove.com/v4/access_token?grant_type=client_credentials';
  const clientIdSecret = `${brightcoveClientId}:${brightcoveClientSecret}`;

  return fetch(url, {
    method: 'POST',
    headers: {
      'content-type': 'application/x-www-form-urlencoded',
      Authorization: `Basic ${base64Encode(clientIdSecret)}`,
    },
  }).then(resolveJsonOrRejectWithError);
}

export const fetchVideoMeta = embed =>
  getAccessToken()
    .then(accessToken => fetchVideo(embed.videoid, embed.account, accessToken))
    .then(resolveJsonOrRejectWithError)
    .then(video => ({ ...embed, video }));
