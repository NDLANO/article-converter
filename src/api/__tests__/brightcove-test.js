import { getContributorGroups } from '../brightcove';

test('parses contributor string correctly', () => {
  const fields = {
    licenseinfo: 'Opphavsmann: Senter for nye medier, Høgskolen i Bergen',
    licenseinfo1: 'Opphavsmann:Adalia film & media',
    licenseinfo2: 'Leverandør:    Norsk Helseinformatikk',
    licenseinfo3: 'Distributør : ndla.no',
    yt_privacy_status: 'private',
    license: 'Navngivelse-Del på samme vilkår',
  };

  const authors = getContributorGroups(fields);
  expect(authors).toMatchSnapshot();
});

test('uses fallback if contributor string could not be parsed', () => {
  const fields = {
    licenseinfo: 'Opphavsmann: Senter for nye medier, Høgskolen i Bergen',
    licenseinfo1: 'Opphavsmann Adalia film & media',
    yt_privacy_status: 'private',
    license: 'Navngivelse-Del på samme vilkår',
  };

  const authors = getContributorGroups(fields);
  expect(authors).toMatchSnapshot();
});
