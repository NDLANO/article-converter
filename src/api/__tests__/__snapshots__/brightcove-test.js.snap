// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parses contributor string correctly 1`] = `
Object {
  "creators": Array [
    Object {
      "name": "Senter for nye medier, Høgskolen i Bergen",
      "type": "originator",
    },
    Object {
      "name": "Adalia film & media",
      "type": "originator",
    },
  ],
  "processors": Array [],
  "rightsholders": Array [
    Object {
      "name": "Norsk Helseinformatikk",
      "type": "supplier",
    },
    Object {
      "name": "ndla.no",
      "type": "distributor",
    },
  ],
}
`;

exports[`uses fallback if contributor string could not be parsed 1`] = `
Object {
  "creators": Array [
    Object {
      "name": "Senter for nye medier, Høgskolen i Bergen",
      "type": "originator",
    },
    Object {
      "name": "Opphavsmann Adalia film & media",
      "type": "",
    },
    Object {
      "name": "Adalia film & media",
      "type": "originator",
    },
    Object {
      "name": "Lucasfilm",
      "type": "",
    },
  ],
  "processors": Array [],
  "rightsholders": Array [],
}
`;
